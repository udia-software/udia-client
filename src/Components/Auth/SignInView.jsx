import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import {
  GridLoadingOverlay,
  StyledAuthButton,
  StyledAuthContainer,
  StyledAuthForm,
  StyledAuthFormContent,
  StyledAuthFormField,
  StyledAuthFormFieldErrors,
  StyledAuthFormFieldset,
  StyledInput
} from "Components/Styled";

const SignInView = ({
  loading,
  email,
  password,
  errors,
  emailErrors,
  passwordErrors,
  handleChangeEmail,
  handleChangePassword,
  handleSubmit
}) => {
  return (
    <StyledAuthContainer>
      <GridLoadingOverlay gridAreaName="form" loading={loading} />
      <h1>Sign In</h1>
      <StyledAuthForm onSubmit={handleSubmit}>
        <StyledAuthFormFieldset>
          <legend>Welcome back, User.</legend>
          <StyledAuthFormFieldErrors errors={errors} />
          <StyledAuthFormContent>
            <StyledAuthFormField>
              <label htmlFor="email">Email:</label>
              <StyledInput
                type="email"
                id="email"
                placeholder="your@email.tld"
                onChange={handleChangeEmail}
                value={email}
              />
              <StyledAuthFormFieldErrors errors={emailErrors} />
            </StyledAuthFormField>
            <StyledAuthFormField>
              <label htmlFor="pw">Password:</label>
              <StyledInput
                type="password"
                id="pw"
                placeholder="••••••••"
                onChange={handleChangePassword}
                value={password}
              />
              <StyledAuthFormFieldErrors errors={passwordErrors} />
            </StyledAuthFormField>
            <StyledAuthButton>Sign In</StyledAuthButton>
          </StyledAuthFormContent>
        </StyledAuthFormFieldset>
      </StyledAuthForm>
      <Link to="/sign-up">I don't have an account →</Link>
      <Link to="/forgot_password">I forgot my password.</Link>
    </StyledAuthContainer>
  );
};

SignInView.propTypes = {
  loading: PropTypes.bool.isRequired,
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  errors: PropTypes.arrayOf(PropTypes.string).isRequired,
  emailErrors: PropTypes.arrayOf(PropTypes.string).isRequired,
  passwordErrors: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleChangeEmail: PropTypes.func.isRequired,
  handleChangePassword: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export { SignInView };
export default SignInView;
