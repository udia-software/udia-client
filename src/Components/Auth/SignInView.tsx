import React, { ChangeEventHandler, FormEventHandler } from "react";
import { Link } from "react-router-dom";
import styled from "../AppStyles";
import Button from "../PureHelpers/Button";
import FormField from "../PureHelpers/FormField";
import FormFieldErrors from "../PureHelpers/FormFieldErrors";

export interface IProps {
  loading: boolean;
  loadingText?: string;
  email: string;
  password: string;
  errors: string[];
  emailErrors: string[];
  passwordErrors: string[];
  handleChangeEmail: ChangeEventHandler;
  handleChangePassword: ChangeEventHandler;
  handleSubmit: FormEventHandler;
}

const SignInContainer = styled.div`
  display: grid;
  grid-template-areas:
    "title"
    "form"
    "silinks";
  grid-auto-rows: auto;
  place-items: center;
  align-content: center;
`;

const SignInTitle = styled.h1`
  grid-area: title;
`;

const FormContainer = styled.form`
  grid-area: form;
`;

const SignInLinks = styled.div`
  grid-area: silinks;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  text-align: center;
  padding-top: 0.5em;
  &>a {
    padding-top: 0.5em;
  }
`;

const FormContent = styled.div`
  display: flex;
  height: 100%;
  flex-direction: column;
  justify-content: space-evenly;
`;

const FormInput = styled.input`
  flex: 1 0 auto;
  width: auto;
  padding: 0.5em;
  font-size: medium;
  background: ${props => props.theme.inputBaseBackgroundColor};
  border: 1px solid ${props => props.theme.inverseColor};
  border-radius: 3px;
  :focus {
    outline: 1px solid ${props => props.theme.purple};
    border: 1px solid ${props => props.theme.purple};
  }
`;

const SignInView = ({
  loading,
  loadingText,
  email,
  password,
  errors,
  emailErrors,
  passwordErrors,
  handleChangeEmail,
  handleChangePassword,
  handleSubmit
}: IProps) => (
  <SignInContainer>
    <SignInTitle>Sign In</SignInTitle>
    <FormContainer onSubmit={handleSubmit}>
      <fieldset>
        <legend>Welcome back, User.</legend>
        <FormFieldErrors errors={errors} />
        <FormContent>
          <FormField error={emailErrors.length > 0}>
            <label htmlFor="email">
              Email:
              <FormInput
                type="email"
                id="email"
                placeholder="your@email.tld"
                onChange={handleChangeEmail}
                value={email}
              />
            </label>
            <FormFieldErrors errors={emailErrors} />
          </FormField>
          <FormField error={passwordErrors.length > 0}>
            <label htmlFor="pw">
              Password:
              <FormInput
                type="password"
                id="pw"
                placeholder="••••••••"
                onChange={handleChangePassword}
                value={password}
              />
            </label>
            <FormFieldErrors errors={passwordErrors} />
          </FormField>
          <Button>Sign In</Button>
        </FormContent>
      </fieldset>
    </FormContainer>
    <SignInLinks>
      <Link to="/sign-up">I don&apos;t have an account →</Link>
      <Link to="/forgot-password">I forgot my password.</Link>
    </SignInLinks>
  </SignInContainer>
);

export default SignInView;
