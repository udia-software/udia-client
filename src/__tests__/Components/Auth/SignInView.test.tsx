import { library } from "@fortawesome/fontawesome-svg-core";
import { faEye } from "@fortawesome/free-solid-svg-icons";
import React from "react";
import { MemoryRouter } from "react-router";
import renderer from "react-test-renderer";
import SignInView from "../../../Components/Auth/SignInView";

beforeAll(() => {
  library.add(faEye);
});

afterAll(() => {
  library.reset();
});

it("renders ForgotPasswordView correctly", () => {
  const stubHandleMethod = () => null;
  const tree = renderer
    .create(
      <MemoryRouter>
        <SignInView
          loading={false}
          email=""
          password=""
          errors={[]}
          emailErrors={[]}
          passwordErrors={[]}
          showPassword={false}
          handleChangeEmail={stubHandleMethod}
          handleChangePassword={stubHandleMethod}
          handleTogglePassword={stubHandleMethod}
          handleSubmit={stubHandleMethod}
        />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
