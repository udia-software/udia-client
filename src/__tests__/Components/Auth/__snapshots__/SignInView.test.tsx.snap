// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ForgotPasswordView correctly 1`] = `
<div
  className="sc-bZQynM epQqBf"
>
  <div
    className="sc-bxivhb hjJIta"
  >
    <svg
      className="sc-htpNat ctIEmo"
      height="80px"
      version="1.1"
      viewBox="0 0 90 90"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
      y="0px"
    >
      <polygon
        className={undefined}
        points="38.182,57.753 20.18,47.91 20.18,42.455 38.182,32.652 38.182,39.074 25.625,45.113 38.182,51.379"
      />
      <polygon
        className={undefined}
        points="39.848,62.08 46.351,27.918 50.136,27.918 43.56,62.08"
      />
      <polygon
        className={undefined}
        points="51.8,57.78 51.8,51.4 64.372,45.181 51.8,39.028 51.8,32.696 69.82,42.5 69.82,47.91"
      />
      <path
        className={undefined}
        d="M44.999,86.031L9.465,65.517V24.484L44.999,3.969l35.536,20.516v41.029L44.999,86.031ZM13.07,63.434l31.929,18.434L76.93,63.434V26.566L44.999,8.131L13.07,26.565V63.434L13.07,63.434Z"
      />
    </svg>
    Loading...
  </div>
  <h1
    className="sc-gzVnrw hvNDye"
  >
    Sign In
  </h1>
  <form
    className="sc-htoDjs byhNZN"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend
        className="sc-dnqmqq bxVEtg"
      >
        Welcome back, User.
      </legend>
      <ul
        className="sc-bwzfXH jccCNJ"
      />
      <div
        className="sc-gZMcBi gjMTpk"
      >
        <div
          className="sc-fjdhpX bhReIE"
        >
          <label
            htmlFor="email"
          >
            Email:
            <input
              className="sc-gqjmRU fmxPXL"
              id="email"
              onChange={[Function]}
              placeholder="your@email.tld"
              type="email"
              value=""
            />
          </label>
          <ul
            className="sc-bwzfXH jccCNJ"
          />
        </div>
        <div
          className="sc-fjdhpX bhReIE"
        >
          <label
            htmlFor="pw"
          >
            <span
              className="sc-jTzLTM gArCnl"
            >
              <span>
                Password:
              </span>
              <a
                className="sc-EHOje bLbaUl"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye fa-w-18 "
                  data-icon="eye"
                  data-prefix="fas"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M569.354 231.631C512.969 135.949 407.81 72 288 72 168.14 72 63.004 135.994 6.646 231.631a47.999 47.999 0 0 0 0 48.739C63.031 376.051 168.19 440 288 440c119.86 0 224.996-63.994 281.354-159.631a47.997 47.997 0 0 0 0-48.738zM288 392c-75.162 0-136-60.827-136-136 0-75.162 60.826-136 136-136 75.162 0 136 60.826 136 136 0 75.162-60.826 136-136 136zm104-136c0 57.438-46.562 104-104 104s-104-46.562-104-104c0-17.708 4.431-34.379 12.236-48.973l-.001.032c0 23.651 19.173 42.823 42.824 42.823s42.824-19.173 42.824-42.823c0-23.651-19.173-42.824-42.824-42.824l-.032.001C253.621 156.431 270.292 152 288 152c57.438 0 104 46.562 104 104z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 
                Show Password
              </a>
            </span>
            <input
              className="sc-gqjmRU fmxPXL"
              id="pw"
              onChange={[Function]}
              placeholder="••••••••"
              type="password"
              value=""
            />
          </label>
          <ul
            className="sc-bwzfXH jccCNJ"
          />
        </div>
        <button
          className="sc-bdVaJa dDhRHz"
        >
          Sign In
        </button>
      </div>
    </fieldset>
  </form>
  <div
    className="sc-iwsKbI gnZiZr"
  >
    <a
      className="sc-ifAKCX drsjAW"
      href="/sign-up"
      onClick={[Function]}
    >
      I don't have an account →
    </a>
    <a
      className="sc-ifAKCX drsjAW"
      href="/forgot-password"
      onClick={[Function]}
    >
      I forgot my password.
    </a>
  </div>
</div>
`;
